### PR TITLE
Display entitlement key in symbol reference

### PR DESCRIPTION
**Pull request by sofiaromorales**
Bug/issue #, if applicable:  129540266

## Summary

Property list keys that define a human readable title (https://github.com/swiftlang/swift-docc/pull/987) should show the key name under the abstract when referenced form another symbol.

This pr adds the implementation to get this working by passing the property list key information to the topic render reference.

## Dependencies

N/A

## Testing

Steps:
1. Preview the attached docc archive.
2. Assert that in the top level page, the foo symbol displays the key name under the abstract section.

[plistkey.docc.zip](https://github.com/user-attachments/files/16849526/plistkey.docc.zip)

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- [X] Added tests
- [X] Ran the `./bin/test` script and it succeeded
- [ ] Updated documentation if necessary

### CODE REVIEW

File: Sources/SwiftDocC/Model/Rendering/DocumentationContentRenderer.swift [hunk 1]
```diff
@@ -363,7 +363,17 @@ public class DocumentationContentRenderer {
...
+        
+        // Add key information for property lists.
+        var propertyListKeyNames: TopicRenderReference.PropertyListKeyNames? = nil
+        if let nodePropertyListDetails = node?.symbol?.plistDetails  {
+            propertyListKeyNames = TopicRenderReference.PropertyListKeyNames(
+                titleStyle: (nodePropertyListDetails.customTitle != nil) ? .useDisplayName : .useRawKey,
+                rawKey: nodePropertyListDetails.rawKey,
+                displayName: nodePropertyListDetails.customTitle
+            )
+        }
```
Comment (d-ronnqvist) [inline Sources/SwiftDocC/Model/Rendering/DocumentationContentRenderer.swift:375]: FYI: this pattern of 

```swift
var someValue: Something? = nil
if let somethingOptional {
   someValue = // create a `Something` based on the unwrapped `somethingOptional`
}
```

or

```swift
let someValue: Something?
if let somethingOptional {
   someValue = // create a `Something` based on the unwrapped `somethingOptional`
} else {
   someValue = nil 
}
```

Is good use of [`Optional.map(_:)`](https://developer.apple.com/documentation/swift/optional/map(_:)), which turns the pattern into 

```swift
let someValue = somethingOptional.map {
   // create a `Something` based on the unwrapped `$0`
}
```


```suggestion
        let propertyListKeyNames = node?.symbol?.plistDetails.map {
            TopicRenderReference.PropertyListKeyNames(
                titleStyle: ($0.customTitle != nil) ? .useDisplayName : .useRawKey,
                rawKey: $0.rawKey,
                displayName: $0.customTitle
            )
        }
```

Comment (sofiaromorales) [reply Sources/SwiftDocC/Model/Rendering/DocumentationContentRenderer.swift]: Resolved https://github.com/swiftlang/swift-docc/pull/1018/commits/d20f1ee1bfedc2d0765b9e6f22579d321b35ccb9

**Review by d-ronnqvist — APPROVED** [review]
LGTM

**Comment by sofiaromorales** [comment]
@swift-ci please test

**Pull request merged**